src/wallet-account.ts

```diff
@@ -235,7 +235,10 @@ class ConnectedWalletAccount extends Account {
                     return false;
                 }
                 const [{ functionCall }] = actions;
-                return functionCall && (allowedMethods.length === 0 || allowedMethods.includes(functionCall.methodName));
+                return functionCall &&
+                    functionCall.deposit.toString() === "0" && // TODO: Should support charging amount smaller than allowance?
```

Comment:
This now expects `deposit` to be defined. I don't think we required that before, and now any contract calls that only passed one or two arguments are going to start breaking. Is that accurate, @vgrichina ?

Reply:
Thanks for catching this, fixed in follow-up!